### ISSUE TITLE
make error unimplemented: -fexcess-precision=standard for C++

### ISSUE THREAD

**Issue by digoal** [jaiminpan/pg_jieba#1]
```
root@digoal-> make
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -O2 -DLOCK_DEBUG -DBTREE_BUILD_STATS -DWAL_DEBUG -DOPTIMIZER_DEBUG -DGEQO_DEBUG -DCOPY_PARSE_PLAN_TREES -DTRACE_SYNCSCAN -fpic -I. -I. -I../../src/include -D_GNU_SOURCE -I/usr/include/libxml2   -c -o pg_jieba.o pg_jieba.c
g++ -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -O2 -DLOCK_DEBUG -DBTREE_BUILD_STATS -DWAL_DEBUG -DOPTIMIZER_DEBUG -DGEQO_DEBUG -DCOPY_PARSE_PLAN_TREES -DTRACE_SYNCSCAN -fpic -I. -I. -I../../src/include -D_GNU_SOURCE -I/usr/include/libxml2  -DLOGGER_LEVEL=LL_WARN -o jieba.o -c libjieba/jieba.cpp
cc1plus: warning: command line option ‘-Wmissing-prototypes’ is valid for C/ObjC but not for C++
cc1plus: warning: command line option ‘-Wdeclaration-after-statement’ is valid for C/ObjC but not for C++
cc1plus: sorry, unimplemented: -fexcess-precision=standard for C++
make: *** [jieba.o] Error 1
```

**Comment by jaiminpan** [comment]
what's your configure?

**Comment by jaiminpan** [comment]
试一试下面这个新的makefile配置
```
# contrib/pg_jieba/Makefile

MODULE_big = pg_jieba
OBJS = pg_jieba.o \
		jieba.o

EXTENSION = pg_jieba
DATA = pg_jieba--1.0.sql pg_jieba--unpackaged--1.0.sql
DATA_TSEARCH = dict/jieba.dict.utf8 dict/jieba.hmm_model.utf8 dict/jieba.user.dict.utf8

REGRESS = pg_jieba

SHLIB_LINK = -lstdc++

ifdef USE_PGXS
PG_CONFIG = pg_config
PGXS := $(shell $(PG_CONFIG) --pgxs)
include $(PGXS)
else
subdir = contrib/pg_jieba
top_builddir = ../..
include $(top_builddir)/src/Makefile.global
include $(top_srcdir)/contrib/contrib-global.mk
endif

CFLAGS_JB := $(filter-out -Wmissing-prototypes -Wdeclaration-after-statement -fexcess-precision=standard, $(CFLAGS))

jieba.o: libjieba/jieba.cpp
	g++ $(CFLAGS_JB) $(CPPFLAGS) -DLOGGER_LEVEL=LL_WARN -o $@ -c $^

```

**Comment by digoal** [comment]
ok now, but:
postgres=# create extension pg_jieba;
ERROR:  could not load library "/opt/pgsql9.4.4/lib/pg_jieba.so": /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /opt/pgsql9.4.4/lib/pg_jieba.so)

**Comment by digoal** [comment]
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -O2 -DLOCK_DEBUG -DBTREE_BUILD_STATS -DWAL_DEBUG -DOPTIMIZER_DEBUG -DGEQO_DEBUG -DCOPY_PARSE_PLAN_TREES -DTRACE_SYNCSCAN -fpic -I. -I. -I../../src/include -D_GNU_SOURCE -I/usr/include/libxml2   -c -o pg_jieba.o pg_jieba.c
g++ -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -g -O2 -DLOCK_DEBUG -DBTREE_BUILD_STATS -DWAL_DEBUG -DOPTIMIZER_DEBUG -DGEQO_DEBUG -DCOPY_PARSE_PLAN_TREES -DTRACE_SYNCSCAN -fpic -I. -I. -I../../src/include -D_GNU_SOURCE -I/usr/include/libxml2  -DLOGGER_LEVEL=LL_WARN -o jieba.o -c libjieba/jieba.cpp
In file included from libjieba/jieba/MPSegment.hpp:7:0,
                 from libjieba/jieba/MixSegment.hpp:5,
                 from libjieba/jieba.cpp:15:
libjieba/jieba/limonp/Logger.hpp: In static member function ‘static void limonp::Logger::LoggingF(size_t, const char*, int, const char*, ...)’:
libjieba/jieba/limonp/Logger.hpp:52:59: warning: function might be possible candidate for ‘gnu_printf’ format attribute [-Wsuggest-attribute=format]
       int n = vsnprintf((char *)msg.c_str(), size, fmt, ap);
                                                           ^
libjieba/jieba/limonp/Logger.hpp:52:59: warning: function might be possible candidate for ‘gnu_printf’ format attribute [-Wsuggest-attribute=format]
In file included from libjieba/jieba/DictTrie.hpp:11:0,
                 from libjieba/jieba/MPSegment.hpp:8,
                 from libjieba/jieba/MixSegment.hpp:5,
                 from libjieba/jieba.cpp:15:
libjieba/jieba/limonp/StringUtil.hpp: In function ‘std::string limonp::string_format(const char*, ...)’:
libjieba/jieba/limonp/StringUtil.hpp:35:57: warning: function might be possible candidate for ‘gnu_printf’ format attribute [-Wsuggest-attribute=format]
     int n = vsnprintf((char *)str.c_str(), size, fmt, ap);
                                                         ^
libjieba/jieba/limonp/StringUtil.hpp:35:57: warning: function might be possible candidate for ‘gnu_printf’ format attribute [-Wsuggest-attribute=format]
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -O2 -DLOCK_DEBUG -DBTREE_BUILD_STATS -DWAL_DEBUG -DOPTIMIZER_DEBUG -DGEQO_DEBUG -DCOPY_PARSE_PLAN_TREES -DTRACE_SYNCSCAN -fpic -shared -o pg_jieba.so pg_jieba.o jieba.o -L../../src/port -L../../src/common -Wl,--as-needed -Wl,-rpath,'/opt/pgsql9.4.4/lib',--enable-new-dtags  -lstdc++ 
root@digoal-> make install
/bin/mkdir -p '/opt/pgsql9.4.4/lib'
/bin/mkdir -p '/opt/pgsql9.4.4/share/extension'
/bin/mkdir -p '/opt/pgsql9.4.4/share/extension'
/bin/mkdir -p '/opt/pgsql9.4.4/share/tsearch_data'
/usr/bin/install -c -m 755  pg_jieba.so '/opt/pgsql9.4.4/lib/pg_jieba.so'
/usr/bin/install -c -m 644 pg_jieba.control '/opt/pgsql9.4.4/share/extension/'
/usr/bin/install -c -m 644 pg_jieba--1.0.sql pg_jieba--unpackaged--1.0.sql '/opt/pgsql9.4.4/share/extension/'
/usr/bin/install -c -m 644 dict/jieba.dict.utf8 dict/jieba.hmm_model.utf8 dict/jieba.user.dict.utf8 '/opt/pgsql9.4.4/share/tsearch_data/'

**Comment by jaiminpan** [comment]
what's your linux version?
is it the same env between make pg and running pg??
it seems the make env and running env using diffenent stdc++ library

2015-12-03 17:25 GMT+08:00 Digoal.zhou <notifications@github.com>:

**Comment by eitazhou** [comment]
under MAC OS X (10.11),  the last line of Makefile should be 
clang++  $(CFLAGS_JB) $(CPPFLAGS)  -std=c++11 -stdlib=libc++  -DLOGGER_LEVEL=LL_WARN -o $@ -c $^

**Comment by jaiminpan** [comment]
@eitazhou 
I try to fix it by commit c40575b45f1b32eebe3604204f5aba6c62b56abf
Because my MacOS(10.11) use gcc and g++ as default compiler and it works.

'-std=c++11 -stdlib=libc++'  is not added because I am not sure if is a necessary part.
Please help check if tie fix works with clang++ as default compiler.
Thank you for your information.
